### DISCUSSION TITLE
steering committee opinion needed: $ui5(selector) as alias for browser.asControl(selector)?

### DISCUSSION THREAD

**Discussion by vobu** [ui5-community/wdi5#374] [Ideas]
Hi colleagues,

recently some v-Coffee-Hall™ discussions happened around the primary entry into the `wdi5` API: `browser.asControl()`.
The original (and still valid) notion was to be DevX-compatible with UIveri5's `element(by.control(selector))`.

Yet there is `wdio`'s
- `$(...)` shorthand for `browser.findElement(...)` https://webdriver.io/docs/api/element/$
- `$(selector).react$()` for querying React Components https://webdriver.io/docs/api/element/react$
- `$(selector).shadow$(selector)` for getting into the shadow DOM of web components https://webdriver.io/docs/api/element/shadow$

Now here's the q: 
Should we do `ui5$(selector)` or `$ui5(selector)` for `browser.asControl(selector)`?

Thanks for your opinions @ArnaudBuchholz @dominikfeininger @nicoschoenteich  @monavari-lebrecht @hmanchev @aiopa @SebastianWolf-SAP !

**Comment by monavari-lebrecht** [comment]
I dont like globals that much ... What about putting the line 
`export const $ui5 = browser.asControl`
into src/index.ts 

So, when the user starts typing $ui5 vscode imports it automatically by adding import { $ui5 } from "wdio-ui5-service"

For the programmer it is no real difference for the shorthand but we avoid the global.

**Comment by aiopa** [comment]
ui5$(selector), so that we follow what looks like a standard placement for $.

Personally i dont mind a global or not global short hand selector. The only comment on that would be that if you have worked with react$ and it is global, it may be a bit confusing that ui5$ is not. I find it more important to follow the standards that are set.

**Reply by monavari-lebrecht** [reply]
Consistency is a good point and I agree with you and @vobu that this is a good way.

what about a second shorthand $$ui5 für allControls?
